db/sql/deploy/buildings/functions/add_update_facilities_attributes_function.sql

```diff
@@ -0,0 +1,367 @@
+        )
+        -- Filter buildings to those with more than half the geometry within facility polygon
+        SELECT *
+        FROM bo_intersects_fac
+        WHERE bo_intersect_ratio > 0.5
+    ),
+    -- Join building use table using LEFT JOIN because building use will be empty
+    building_use_joined AS (
+    SELECT
+        bo_in_fac.fac_use,
+        bo_in_fac.building_id,
+        bu.building_use_id,
+        u.value
+    FROM bo_in_fac
+    LEFT JOIN buildings.building_use bu USING (building_id)
+    LEFT JOIN buildings.use u USING (use_id)
+    WHERE bu.building_use_id is NULL
+    ),
+    -- Add new building use 
+    inserted AS (
+        INSERT INTO buildings.building_use (building_use_id, building_id, use_id, begin_lifespan)
+        SELECT nextval('buildings.building_use_building_use_id_seq'),
+            building_use_joined.building_id,
+            (SELECT use_id FROM buildings.use WHERE use.value = building_use_joined.fac_use),
+            NOW()
+        FROM building_use_joined
+    )
+    -- Return count for reporting 
+    SELECT count(*)::integer FROM building_use_joined;
+$$
+LANGUAGE sql VOLATILE;
+
+
+------------------------------------------------------------------------------
+-- BUILDING NAME/USE REMOVE
+-- Where building has NAME/USE with no matching facility polygon, remove NAME/USE.
+-- Only remove NAME/USE for USE types included in NZ Facilities, currently
+-- these are Hospitals and Schools.  Other USE types do not have corresponding
+-- facility polygons to compare, so removal of NAME and USE needs to be in the same
+-- function.
+------------------------------------------------------------------------------
+CREATE OR REPLACE FUNCTION buildings.update_facilities_name_use_remove()
+RETURNS integer AS
+$$
+    -- Select buildings with a use that is included in NZ Facilities, excluding retired uses and retired buildings
+    WITH bo_has_use AS (
+        SELECT
+            building_id,
+            bn.building_name_id AS bn_building_name_id,
+            bn.building_name AS bn_building_name,
+            bu.building_use_id AS bu_building_use_id,
+            u.value AS u_value,
+            bo.shape
+        FROM buildings.building_outlines bo
+        JOIN buildings.building_name bn USING (building_id)
+        JOIN buildings.building_use bu USING (building_id)
+        JOIN buildings.use u USING (use_id)
+        WHERE u.use_id IN (16,27) -- use_id 16 = Hospital, 27 = School
+            AND bu.end_lifespan IS NULL
+            AND bo.end_lifespan IS NULL
+    ),
+    -- Join facility polygons using intersects and LEFT JOIN because we want to include those with no intersect
+    bo_outside_fac AS (
+    SELECT 
+        building_id,
+        bn_building_name_id,
+        bn_building_name,
+        bu_building_use_id,
+        u_value,
+        bo.shape
+    FROM bo_has_use bo
+    LEFT JOIN buildings_reference.nz_facilities f
+    ON ST_Intersects(bo.shape, f.shape)
+    GROUP BY building_id,
```

Comment:
took me a second to figure out why the group by, but this is very clever way to handle buildings which weave in and out of a facility